vercel/src/lib/AsciiPreview.svelte

```diff
@@ -0,0 +1,77 @@
+<script lang="ts">
+  import Chip from './Chip.svelte';
+
+  export let name: string;
+  export let ansi: string[];
+  export let hex: string[] | null = null;
+  export let chip: string;
+  export let ascii: string = '';
+
+  let dark = true;
+  let trueColor = hex != null;
+
+  $: html = ascii
+    .split('\n')
+    .map((line) => {
+      // TODO Clean up, this is hard to read
+      let spanCount = 0;
+      const htmlLine = line.replace(/\{(\d+)\}/g, (_match, index) => {
+        const i = Number.parseInt(index, 10);
+        const spanText = trueColor
+          ? `<span style="color: ${hex[i]}">`
+          : `<span style="color: ${mapToDefaultTerminalFgColor(
+              ansi[i],
+              dark
+            )}">`;
+        spanCount++;
+        return spanText;
+      });
+      return `${htmlLine}${'</span>'.repeat(spanCount)}`;
+    })
+    .join('\n');
+
+  function mapToDefaultTerminalFgColor(color: string, dark: boolean): string {
+    return color === 'white' && !dark ? 'black' : color;
+  }
+</script>
+
+<div class="title-row">
+  <div class="language-name">
+    <Chip color={chip} width={24} height={24} />
+    <h3>{name}</h3>
+  </div>
+  <div>
+    <input type="checkbox" bind:checked={dark} />
+    Dark
```

Comment:
Just a minor nitpick:

I just realized that, now that the `<input>` isn't wrapped in a `<label>`, clicking on the text "Dark" doesn't do anything (previously it would check/uncheck the box just like clicking on the input itself). If we can't use a wrapping `<label>`, perhaps `<label for="dark-checkbox-{index}">` + `<input id="dark-checkbox-{index}">` (this would accomplish the same behavior)?

Reply:
Done in https://github.com/o2sh/onefetch/pull/701/commits/558ccdbc0950502fd349b7905e4a96d63612187f and https://github.com/o2sh/onefetch/pull/701/commits/b64d590a67c6a13d683b72b8e81df22124432655